internal/server/server.go

```diff
@@ -2,105 +2,106 @@ package server
-		Port: cfg.Port,
-	}
-	grpcOpts := server.WithMuxGRPCServerOptions(grpc.UnaryInterceptor(grpc_middleware.ChainUnaryServer(
-		grpc_recovery.UnaryServerInterceptor(),
-		grpc_ctxtags.UnaryServerInterceptor(),
-		grpc_zap.UnaryServerInterceptor(logger),
-		nrgrpc.UnaryServerInterceptor(nr),
-	)))
-
-	muxServer, err := server.NewMux(serverCfg, grpcOpts)
-	if err != nil {
-		return err
-	}
+const defaultGracePeriod = 5 * time.Second
 
-	gw, err := server.NewGateway(serverCfg.Host, serverCfg.Port)
-	if err != nil {
-		return err
+func Serve(ctx context.Context, addr string, logger *zap.Logger, resourceSvc handlersv1.ResourceService) error {
+	grpcOpts := []grpc.ServerOption{
+		grpc.UnaryInterceptor(grpc_middleware.ChainUnaryServer(
+			grpc_recovery.UnaryServerInterceptor(),
+			grpc_ctxtags.UnaryServerInterceptor(),
+			grpc_zap.UnaryServerInterceptor(logger),
+		)),
+		grpc.StatsHandler(&ocgrpc.ServerHandler{}),
 	}
+	grpcServer := grpc.NewServer(grpcOpts...)
+	rpcHTTPGateway := runtime.NewServeMux()
+	reflection.Register(grpcServer)
 
-	err = gw.RegisterHandler(ctx, commonv1.RegisterCommonServiceHandlerFromEndpoint)
-	if err != nil {
+	commonServiceRPC := common.New(version.GetVersionAndBuildInfo())
+	grpcServer.RegisterService(&commonv1.CommonService_ServiceDesc, commonServiceRPC)
+	if err := commonv1.RegisterCommonServiceHandlerServer(ctx, rpcHTTPGateway, commonServiceRPC); err != nil {
 		return err
 	}
 
-	err = gw.RegisterHandler(ctx, entropyv1beta1.RegisterResourceServiceHandlerFromEndpoint)
-	if err != nil {
+	resourceServiceRPC := handlersv1.NewAPIServer(resourceSvc)
+	grpcServer.RegisterService(&entropyv1beta1.ResourceService_ServiceDesc, resourceServiceRPC)
+	if err := entropyv1beta1.RegisterResourceServiceHandlerServer(ctx, rpcHTTPGateway, resourceServiceRPC); err != nil {
 		return err
 	}
 
-	muxServer.SetGateway("/api", gw)
-	muxServer.RegisterHandler("/ping", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		_, _ = fmt.Fprintf(w, "pong")
+	httpRouter := gorillamux.NewRouter()
+	httpRouter.PathPrefix("/api/").Handler(http.StripPrefix("/api", rpcHTTPGateway))
+	httpRouter.Handle("/ping", http.HandlerFunc(func(wr http.ResponseWriter, req *http.Request) {
+		_, _ = fmt.Fprintf(wr, "pong")
 	}))
 
-	muxServer.RegisterService(
-		&commonv1.CommonService_ServiceDesc,
-		common.New(version.GetVersionAndBuildInfo()),
-	)
+	logger.Info("starting server", zap.String("addr", addr))
 
-	v1Handler := handlersv1.NewAPIServer(resourceSvc)
-
-	muxServer.RegisterService(
-		&entropyv1beta1.ResourceService_ServiceDesc,
-		v1Handler,
+	httpRouter.Use(
+		requestID(),
+		withOpenCensus(),
+		requestLogger(logger), // nolint
 	)
+	h := grpcHandlerFunc(grpcServer, httpRouter)
+	return gracefulServe(ctx, logger, defaultGracePeriod, addr, h)
+}
 
-	logger.Info("starting server", zap.String("addr", cfg.addr()))
-	return gracefulServe(ctx, muxServer)
+func grpcHandlerFunc(grpcServer *grpc.Server, otherHandler http.Handler) http.Handler {
```

Comment:
are we not using the salt/mux package ?

Reply:
we have to modify the new mux package again and make a release for that. (because of the cmux issue we found)

Reply:
Got it. Then for now lets keep this.